py-polars/polars/dataframe/frame.py

```diff
@@ -3899,7 +3899,8 @@ def insert_at_idx(self, index: int, series: Series) -> Self:
 
     def filter(
         self,
-        predicate: (Expr | str | Series | list[bool] | np.ndarray[Any, Any] | bool),
+        *predicates: IntoExpr | list[bool] | np.ndarray[Any, Any],
```

Comment:
We now have `IntoExprColumn` for cases like this - this way, we don't allow things like datetimes:

```suggestion
        *predicates: IntoExprColumn | bool | list[bool] | np.ndarray[Any, Any],
```